### PR TITLE
Mongo db implementation

### PR DESCRIPTION
**Pull request by sumants-dev**

### CODE REVIEW

**Comment by sumants-dev** [comment]
Ok, I think this is in good shape now. I'll see how DocumentDb on AWS goes and get back to you.

**Comment by markwhiting** [comment]
Did you manage to work out the issues you were facing yesterday?

**Comment by sumants-dev** [comment]
Ok. Cool we are now able to connect sucesfully to the docDB on aws. I'll do some testing if it works like the local one.

**Comment by sumants-dev** [comment]
This is ready to go, and has been tested on the aws cluster. Let me know if you want anything else.

File: index.html
Comment (markwhiting) [inline index.html:1]: Is this a fallback?

**Review by markwhiting — APPROVED** [review]
This looks good, though I wonder why so many files have changed, even though they look like nothing has moved at all?

I also note that the way the DB is evoked is a bit counter to my intuitions, e.g., 
`Db_Wrapper.set_collection("responses");` — is this confined to the scope of the calling function, if not, how do we avoid a potential race condition? (i.e., if there are several edit events happening at the same time on different collections).

I assume thats worked out, and if so, approved, but if you're not sure, can you follow up on that.

File: index.html
Comment (sumants-dev) [reply index.html:1]: I had to do a revert  when developing when debugging, so there's a commit that reverted everything back. So it seems like alot of things changed and nothing much changed because of the revert.

The race conditions are prevented because the set_collections just sets the string.
```
  set_collection(collection: string): void {
    this.collection = collection
  }
```
Then when you call a function like find, then it use the set up function, so all transactions occur when function call is invoked.

```
  async set_up() {
    await this.client.connect()
    const database = this.client.db(this.db)
    const collection_obj = database.collection(this.collection)
    return collection_obj
  }

  async find(query: any) {
    const collection_obj = await this.set_up()
...
```

Taking a second look. This seems not very clean.
How about we change the api to ```async find(query: any) TO async find(query: any, collection: string) ```

Comment (sumants-dev) [reply index.html:1]: Not sure what that index.html is doing. I'm pretty sure I can delete it in the next round if you like the api change.

Comment (markwhiting) [reply index.html:1]: Yes, I prefer that API. Let's swap it over before merging.

**Review by markwhiting — CHANGES_REQUESTED** [review]
Just the api adjustment, and as you said the index.html can probably be removed.

**Review by sumants-dev — COMMENTED** [review]
Ok. Did the changes and tested it locally. Should be good to go!

**Review by markwhiting — APPROVED** [review]

**Review by markwhiting — APPROVED** [review]

**Pull request merged**
